src/components/Buttons/ToggleButton/ToggleButton.tsx

```diff
@@ -1,19 +1,40 @@
-import React, { FC } from 'react';
+import { useActionCreators, useAppSelector } from '@hooks/storeHooks/useAppStore';
+import { CurrencyActions } from '@store/UI_store/CurrencySlice/CurrencySlice';
+import React, { FC, useState } from 'react';
 
 //UI
 import classes from './ToggleButton.module.css';
 
 interface ToggleButtonProps {
     isToggle: boolean;
-    onToggle: (e: React.ChangeEvent<HTMLInputElement>) => void
+    onToggle: (e: React.ChangeEvent<HTMLInputElement>) => void;
 }
 const ToggleButton:FC<ToggleButtonProps> = ({isToggle, onToggle}) => {
     return (
         <label className={classes.switch}>
             <input type='checkbox' className={classes.checkbox} checked={isToggle} onChange={onToggle} />
-            <span className={classes.slider}/>
+            <span className={classes.slider} />
         </label>
     );
 };
 
-export default ToggleButton;
\ No newline at end of file
+export default ToggleButton;
+
+
+export const ToggleCurrencyButton: FC = () => {
+    const currency = useAppSelector(state => state.persistedCurrencySlice.currency)
```

Comment:
useAppSelector<Here type of store>(state => state.persistedCurrencySlice.currency);